src/CMakeLists.txt

```diff
@@ -612,7 +612,7 @@ add_executable(ceph-objectstore-tool
   tools/ceph_objectstore_tool.cc
   tools/RadosDump.cc
   $<TARGET_OBJECTS:common_util_obj>)
-target_link_libraries(ceph-objectstore-tool osd os global ${Boost_PROGRAM_OPTIONS_LIBRARY} fuse dl)
+target_link_libraries(ceph-objectstore-tool osd os global ${Boost_PROGRAM_OPTIONS_LIBRARY} ${CMAKE_DL_LIBS})
```

Comment:
On 6-7-2016 10:54, Kefu Chai wrote:
> In src/CMakeLists.txt
> <https://github.com/ceph/ceph/pull/10149#discussion_r69695227>:
> 
>> @@ -612,7 +612,7 @@ add_executable(ceph-objectstore-tool
>>    tools/ceph_objectstore_tool.cc
>>    tools/RadosDump.cc
>>    $<TARGET_OBJECTS:common_util_obj>)
>> -target_link_libraries(ceph-objectstore-tool osd os global ${Boost_PROGRAM_OPTIONS_LIBRARY} fuse dl)
>> +target_link_libraries(ceph-objectstore-tool osd os global ${Boost_PROGRAM_OPTIONS_LIBRARY} ${CMAKE_DL_LIBS})
> 
> ceph-objectstore-tool supports FuseStore if `HAVE_LIBFUSE. in that case
> we need to link against fuse?

I grepped the code, and the only reference was |enable_disable_fuse|,
but there is more HAVE_FUSE parts in src/osd/*
So I guess we'd need to match a WITH_FUSE with the OSD building.

But I'll fix this fuse depedancy for ceph-objectstore-tool in this pull
first.

And the dl <> CMAKE_DL_LIBS fix was just because I was here anyways.
Most of the others are in the FreeBSD cmake changes.
Want me to take 'm out of there and make a separate pull?